packages/jaeger-ui/src/components/TracePage/index.js

```diff
@@ -357,35 +337,32 @@ export class TracePageImpl extends React.PureComponent<TracePageProps, TracePage
     if (trace.state === fetchedState.ERROR || !data) {
       return <ErrorMessage className="ub-m3" error={trace.error || 'Unknown error'} />;
     }
-    const { duration, processes, spans, startTime, traceID } = data;
-    const maxSpanDepth = _maxBy(spans, 'depth').depth + 1;
-    const numberOfServices = new Set(_values(processes).map(p => p.serviceName)).size;
-    const tracePageProps = {
-      duration,
-      maxDepth: maxSpanDepth,
-      name: getTraceName(spans),
-      numServices: numberOfServices,
-      numSpans: spans.length,
+
+    const isEmbedded = Boolean(embedded);
+    const headerProps = {
       slimView,
-      timestamp: startTime,
-      traceID,
-      onSlimViewClicked: this.toggleSlimView,
       textFilter,
-      prevResult: this._scrollManager.scrollToPrevVisibleSpan,
-      nextResult: this._scrollManager.scrollToNextVisibleSpan,
+      viewRange,
+      canCollapse: !embedded || !embedded.timeline.hideSummary || !embedded.timeline.hideMinimap,
       clearSearch: this.clearSearch,
-      resultCount: findMatchesIDs ? findMatchesIDs.size : 0,
-      updateTextFilter: this.updateTextFilter,
-      archiveButtonVisible: archiveEnabled,
+      hideMap: Boolean(embedded && embedded.timeline.hideMinimap),
+      hideSummary: Boolean(embedded && embedded.timeline.hideSummary),
+      linkToStandalone: getUrl(id),
+      nextResult: this._scrollManager.scrollToNextVisibleSpan,
       onArchiveClicked: this.archiveTrace,
+      onSlimViewClicked: this.toggleSlimView,
+      prevResult: this._scrollManager.scrollToPrevVisibleSpan,
       ref: this._searchBar,
-    };
-
-    const tracePageEmbedProps = {
-      searchParams,
-      fromSearch,
-      enableDetails,
-      onGoFullViewClicked: this.goFullView,
+      resultCount: findMatchesIDs ? findMatchesIDs.size : 0,
+      showArchiveButton: !isEmbedded && archiveEnabled,
+      showShortcutsHelp: !isEmbedded,
```

Comment:
to follow the idea of versioned embedding, should these properties be part of the embedded state so that future versions of embedded could affect these separately? i.e.: `showShortCutsHelp = embedded.showShortCutsHelp,`